tests/integ/modin/groupby/test_groupby_apply.py

```diff
@@ -857,6 +858,28 @@ def operation(df):
                 operation,
             )
 
+    # TODO: SNOW-1429855 test support include_groups = False
+    @sql_count_checker(
+        query_count=8,
+        udtf_count=UDTF_COUNT,
+        join_count=JOIN_COUNT,
+    )
+    def test_group_apply_return_df_from_lambda(self):
+        diamonds_path = (
+            pathlib.Path(__file__).parent.parent.parent.parent
+            / "resources"
+            / "diamonds.csv"
+        )
+        diamonds_pd = native_pd.read_csv(diamonds_path)
+        eval_snowpark_pandas_result(
+            pd.DataFrame(diamonds_pd),
+            diamonds_pd,
+            lambda diamonds: diamonds.groupby("cut").apply(
+                lambda x: x.sort_values("price", ascending=False).head(5),
+                include_groups=True,
```

Comment:
looks like this PR is also fixing a bug that `include_groups=True` would raise an exception. Should remove the `include_groups=True` here and have a separate test case for that?

Reply:
I added a todo above to test both True and False in the next PR.